InvenTree/label/api.py

```diff
@@ -201,21 +204,97 @@ def print(self, request, items_to_print):
         label_names = []
         label_instances = []
 
-        # Merge one or more PDF files into a single download
-        for item in items_to_print:
+        if not self.get_object().multipage:
+            # Merge one or more PDF files into a single download
+            for item in items_to_print:
+                label = self.get_object()
+                label.object_to_print = item
+
+                label_name = label.generate_filename(request)
+
+                label_names.append(label_name)
+                label_instances.append(label)
+
+                if debug_mode and plugin is None:
+                    # Note that debug mode is only supported when not using a plugin
+                    outputs.append(label.render_as_string(request))
+                else:
+                    outputs.append(label.render(request))
+        else:
+            # multi page mode -> render into a single HTML output containing a table
             label = self.get_object()
-            label.object_to_print = item
 
-            label_name = label.generate_filename(request)
+            col_count = math.floor(label.page_width / label.width)
+            row_count = math.floor(label.page_height / label.height)
+
+            html_output = '<html><head><style>'
+            html_output += \
+                '@page {size: %dmm %dmm; margin-top: 0mm; margin-bottom: 0mm; margin-left: 0mm; margin-right: 0mm; padding: 0mm;}' \
+                % (label.page_width, label.page_height)
+            html_output += \
+                '.main-table { page-break-after: always; table-layout: fixed; width: %dmm; border-spacing: 0mm 0mm;}' \
+                % (label.page_width)
+            html_output += '.main-table-cell { border-bottom: %s; border-left: %s; width: %dmm; height: %dmm; padding: 0mm; }' \
+                           % (label.multipage_border, label.multipage_border, label.width, label.height)
+            html_output += '.main-table tr:first-child td {border-top: %s}' % label.multipage_border
+            html_output += '.main-table-row> td:last-child {border-right: %s}' % label.multipage_border
+            html_output += 'body {margin: 0mm !important;}'
+            html_output += '</style></head><table class="main-table"><tr class="main-table-row">'
+            col_counter = 0
+            row_counter = 0
+            item_counter = 0
+
+            for item in items_to_print:
```

Comment:
Question: Wouldn't this be easier with an html template?

Reply:
Agreed, can you make a template and throw the page data into the template as a list of "pages"?

```html
{% for page in page %}
<!-- Render page -->
{% endfor %}
```